genevents.go

```diff
@@ -0,0 +1,367 @@
+					Type:    "float32",
+				},
+			},
+		},
+		{
+			Comment: "MouseWheel is a mouse wheel event.",
+			Members: []Member{
+				{
+					Comment: "X is the X position of the mouse wheel. This value is expressed in arbitrary units. It increases when the mouse wheel is scrolled downwards, and decreases when the mouse is scrolled upwards.",
+					Type:    "float32",
+				},
+				{
+					Comment: "Y is the Y position of the mouse wheel. This value is expressed in arbitrary units. It increases when the mouse wheel is scrolled to the right, and decreases when the mouse is scrolled to the left.",
+					Type:    "float32",
+				},
+				{
+					Comment: "DeltaX is the change in X since the last MouseWheel event. This value is expressed in arbitrary units. It is positive when the mouse wheel is scrolled downwards, and negative when the mouse is scrolled upwards.",
+					Type:    "float32",
+				},
+				{
+					Comment: "DeltaY is the change in Y since the last MouseWheel event. This value is expressed in arbitrary units. It is positive when the mouse wheel is scrolled to the right, and negative when the mouse is scrolled to the left.",
+					Type:    "float32",
+				},
+			},
+		},
+		{
+			Comment: "MouseButtonDown is a mouse button press event.",
+			Members: membersMouseButton,
+		},
+		{
+			Comment: "MouseButtonUp is a mouse button release event.",
+			Members: membersMouseButton,
+		},
+		{
+			Comment: "MouseEnter occurs when the mouse enters the view window.",
+			Members: membersMouseEnter,
+		},
+		{
+			Comment: "MouseLeave occurs when the mouse leaves the view window.",
+			Members: membersMouseEnter,
+		},
+		{
+			Comment: "TouchBegin occurs when a touch begins.",
+			Members: membersTouch,
+		},
+		{
+			Comment: "TouchMove occurs when a touch moved, or in other words, is dragged.",
+			Members: membersTouch,
+		},
+		{
+			Comment: "TouchEnd occurs when a touch ends.",
+			Members: membersTouch,
+		},
+		{
+			Comment: "TouchCancel occurs when a touch is canceled. This can happen in various situations, depending on the underlying platform, for example when the aplication loses focus.",
+			Members: []Member{
+				{
+					Comment: "ID identifies the touch that caused the touch event.",
+					Type:    "int",
+				},
+			},
+		},
+		{
+			Comment: "ViewUpdate occurs when the application is ready to update the next frame on the view port.",
+		},
+		{
+			Comment: "ViewSize occurs when the size of the application's view port changes.",
+			Members: []Member{
+				{
+					Comment: "Width is the width of the view. This value is expressed in device independent pixels.",
+					Type:    "int",
+				},
+				{
+					Comment: "Height is the height of the view. This value is expressed in device independent pixels.",
+					Type:    "int",
+				},
+			},
+		},
+	}
+)
+
+const (
+	license = `// Copyright 2013 The Ebiten Authors
```

Comment:
I think legally this should be 2013-2019 to cover the whole development range. But I am not a lawyer.

Reply:
As far as I know, only one year when the file is created should be fine (e.g., Chromium does so). The problem is that auto-generated file cannot determine the year, then I adopted 2013 which Ebiten started.